### PR TITLE
typechecking on client generator

### PR DESCRIPTION
**Pull request by Tcharl**
Second try for https://github.com/jhipster/generator-jhipster/pull/27097
contribute to https://github.com/jhipster/generator-jhipster/issues/26114
---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: jdl/converters/types.d.ts
Comment (mshima) [inline jdl/converters/types.d.ts:1]: We should not import generator/* from jdl/*
JDL should be completely independent.

If converter is too coupled, we should consider moving converters and exporters out of jdl.

File: jdl/jhipster/default-application-options.ts
Comment (mshima) [inline jdl/jhipster/default-application-options.ts:1]: Consider moving this files to generators/base-application/support

File: jdl/converters/types.d.ts
Comment (Tcharl) [reply jdl/converters/types.d.ts:1]: It's the function of converters and exporter to translate from generator spec to jdl spec: make sense to have these objects referencing both concepts

File: jdl/jhipster/default-application-options.ts
Comment (Tcharl) [reply jdl/jhipster/default-application-options.ts:1]: Will be split and spread into their respective generators later on

**Comment by Tcharl** [comment]
Please give more comment or accept: I'm not pushing complete change at once by fear of conflicts or monster reviews. Every pr contributes to an objective and is incremental

**Comment by mshima** [comment]
@Tcharl see https://github.com/jhipster/generator-jhipster/pull/27152, this is caused by circular dependency.
So JDL should be completely independent.
For example we should use generic:
```
Entity A
search A with current (or any)
```

instead of:
```
Entity A
search A with elasticsearch (or couchbase)
```

**Comment by Tcharl** [comment]
So let's fix that glitch.
Leaf gen should point to a base gen (look to be the case, respect!), specific jdl dsl should point to an aggregator (convert/export).
State machine is the core (generator tasks) iterates over contributions (generators implementation). Jdl is like a cli, thus at the early steps of the lifecycle.
To me
- Definition is at the root (object), can be modularized
- command and jdl feed the instances
- generator templates consumes instances. 

There shouldn't be any technological circular dependency but the ones between generators if they are not well designed

**Comment by mshima** [comment]
@Tcharl with https://github.com/jhipster/generator-jhipster/pull/27203 configs added to command.configs will automatically appear on jhipsterConfig and jhipsterConfigWithDefaults.

**Pull request closed without merging**